src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,80 @@ For the full copyright and license information, please view the LICENSE
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
-        {% else %}
-            [{{ key }} => {{ val }}]
-        {% endif %}
-
-        {% if not loop.last and not inline %}
-            <br>
-        {% endif %}
-    {% endfor %}
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
+    {% set options = {
+        'key_translation_domain': false,
+        'value_translation_domain': false,
+        'display': 'both'
+    }|merge(options) %}
+    {%- apply spaceless -%}
+        {%- from _self import render_array -%}
+
+        {%- if not inline -%}
+            <ul>
+        {%- else -%}
+            [
+        {%- endif -%}
+
+        {%- for key, val in value -%}
+            {%- if not inline -%}
+                <li>
+            {%- endif -%}
+
+            {%- if options.display in ['both', 'keys'] -%}
+                {%- if options.key_translation_domain is same as(false) -%}
+                    {{- key -}}
+                {%- else -%}
+                    {%- if options.key_translation_domain is same as(true) -%}
+                        {%- set key_translation_domain = options.default_translation_domain -%}
+                    {%- elseif options.key_translation_domain is same as(null) -%}
+                        {%- set key_translation_domain = null -%}
+                    {%- else -%}
+                        {%- set key_translation_domain = options.key_translation_domain -%}
+                    {%- endif -%}
+
+                    {{- key|trans({}, key_translation_domain) -}}
+                {%- endif -%}
+            {%- endif -%}
+
+            {%- if options.display == 'both' -%}
+                &nbsp;=>&nbsp;
+            {%- endif -%}
+
+            {%- if val is iterable -%}
+                {{ render_array(val, inline, options) }}
+            {%- else -%}
+                {%- if options.display in ['both', 'values'] -%}
+                    {%- if options.value_translation_domain is same as(false) -%}
+                        {{ val }}
+                    {%- else -%}
+                        {%- if options.value_translation_domain is same as(true) -%}
+                            {%- set value_translation_domain = options.default_translation_domain -%}
+                        {%- elseif options.value_translation_domain is same as(null) -%}
+                            {%- set value_translation_domain = null -%}
+                        {%- else -%}
+                            {%- set value_translation_domain = options.value_translation_domain -%}
+                        {%- endif -%}
+
+                        {{- val|trans({}, value_translation_domain) -}}
+                    {%- endif -%}
+                {%- endif -%}
+            {%- endif -%}
+
+            {%- if not inline -%}
+                </li>
+            {%- endif -%}
+
+            {%- if inline == true and not loop.last -%}
+                ,
```

Comment:
If you just remove the space, tests will break. And since you trim, the space is remove.

I don't really know how to get the render `[a=>b, c=>d]`

Reply:
Found the way to fix this. Played with `{%-` and `{%`